### PR TITLE
refactor: show username screen on login

### PR DESCRIPTION
**Pull request by typfel**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Show a screen for picking a username during the login flow if it's not already set. This previously handled using a takeover screen after the login was completed, that screen has now been removed.

Needs releases with:

- [x] https://github.com/wireapp/wire-ios/pull/863

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

#### How to Test

Invite a new user to team and accept the invitation on the web but don't login in the webapp. Instead login via the iOS app and you will see a username screen after you've entered your credentials.

### Attachments

D92A1561-565A-45BB-8CFC-F14BDC81B5DB

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by agisilaos — CHANGES_REQUESTED** [review]
Screenshot 2024-01-24 at 09 50 25

When logging in and during the login flow we implemented a new button (for accessibility purposes) for logging in/continue to next screen etc. Instead of having this arrow button bound to the textfield. Which isn't accessible. 

More on that button can be found in: `SolidButtonDescription` and how it's implemented in variety of places. 

Can you please replace the arrow button with the solid one? Figma designs can be found [here](https://www.figma.com/file/exxrpYZ28N1WOylyw9TYh1/Accessibility-Quick-Wins-(iOS)?type=design&node-id=402-3707&mode=design&t=I5wE32ugiHeqOQJS-0)

**Review by David-Henner — APPROVED** [review]

**Comment by typfel** [comment]
> Screenshot 2024-01-24 at 09 50 25
> When logging in and during the login flow we implemented a new button (for accessibility purposes) for logging in/continue to next screen etc. Instead of having this arrow button bound to the textfield. Which isn't accessible.
> 
> More on that button can be found in: `SolidButtonDescription` and how it's implemented in variety of places.
> 
> Can you please replace the arrow button with the solid one? Figma designs can be found [here](https://www.figma.com/file/exxrpYZ28N1WOylyw9TYh1/Accessibility-Quick-Wins-(iOS)?type=design&node-id=402-3707&mode=design&t=I5wE32ugiHeqOQJS-0)

Ok, I looked into what takes to achieve and I would need to do some big refactoring of the `AuthenticationStepController` and the `ValidatedTextField` to support an external confirm button since as it is now there's no building blocks to achieve this without writing a completely custom view controller for this screen.

I created separate ticket for this: https://wearezeta.atlassian.net/browse/WPB-6292

File: wire-ios-sync-engine/Source/Data Model/ZMUser+Consent.swift [hunk 1]
```diff
@@ -62,13 +62,16 @@ extension ZMUser {
...
+            let apiVersion = BackendInfo.apiVersion,
+            let context = managedObjectContext
+        else {
             return completion(.failure(ConsentRequestError.unknown))
         }
 
         let request = ConsentRequestFactory.fetchConsentRequest(apiVersion: apiVersion)
 
-        request.add(ZMCompletionHandler(on: managedObjectContext!) { response in
+        request.add(ZMCompletionHandler(on: context ) { response in
```
Comment (agisilaos) [inline wire-ios-sync-engine/Source/Data Model/ZMUser+Consent.swift]:
```suggestion
        request.add(ZMCompletionHandler(on: context) { response in
```

**Review by agisilaos — APPROVED** [review]
Looking good!

**Pull request merged**